File: fix_dark_mode.py
import os
import re

def walk(d):
    results = []
    for root, dirs, files in os.walk(d):
        for f in files:
            if f.endswith('.tsx') or f.endswith('.ts'):
                results.append(os.path.join(root, f))
    return results

sysop_dir = '/Users/hidirektor/WebstormProjects/salutbabe-landing/src/app/dashboard/sysop'
files = walk(sysop_dir)

count = 0

def replace_ternary(match):
    light_classes = match.group(2)
    dark_classes = match.group(4)
    
    d_classes = []
    for c in dark_classes.split():
        if not c: continue
        if c.startswith('dark:'):
            d_classes.append(c)
        else:
            d_classes.append('dark:' + c)
            
    return light_classes + ' ' + ' '.join(d_classes)

def replace_inner(match):
    light_classes = match.group(2)
    dark_classes = match.group(4)
    
    d_classes = []
    for c in dark_classes.split():
        if not c: continue
        if c.startswith('dark:'):
            d_classes.append(c)
        else:
            d_classes.append('dark:' + c)
            
    return "'" + light_classes + " " + " ".join(d_classes) + "'"

# Pattern 1: ${theme === 'light' ? 'A' : 'B'}
pattern1 = re.compile(r"\$\{theme === ['\"]light['\"] \? (['\"`])([^'\"`]+)\1 : (['\"`])([^'\"`]+)\3\}")
# Pattern 2: (theme === 'light' ? 'A' : 'B')
pattern2 = re.compile(r"\(theme === ['\"]light['\"] \? (['\"`])([^'\"`]+)\1 : (['\"`])([^'\"`]+)\3\)")

for f in files:
    if 'system-settings' in f or f.endswith('sysop/page.tsx'):
        continue
        
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()
        
    original_content = content
    
    content = pattern1.sub(replace_ternary, content)
    content = pattern2.sub(replace_inner, content)
    
    content = content.replace('dark:bg-surface', 'dark:bg-[#12141C]')
    content = content.replace('dark:bg-[#121214]/60', 'dark:bg-[#12141C]/60')
    content = content.replace('dark:bg-[#1a1a1f]', 'dark:bg-[#1A1D27]')
    
    if content != original_content:
        with open(f, 'w', encoding='utf-8') as file:
            file.write(content)
        print(f"Updated {f}")
        count += 1

print(f"Finished updating {count} files.")

File: fix_dark_mode.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if (file.endsWith('.tsx') || file.endsWith('.ts')) {
                results.push(file);
            }
        }
    });
    return results;
}

const sysopDir = '/Users/hidirektor/WebstormProjects/salutbabe-landing/src/app/dashboard/sysop';
const files = walk(sysopDir);

let count = 0;

for (const file of files) {
    if (file.includes('system-settings') || file.endsWith('sysop/page.tsx')) continue;

    let content = fs.readFileSync(file, 'utf8');
    let originalContent = content;

    // Pattern 1: ${theme === 'light' ? 'A' : 'B'}
    // We will extract A and B, and merge them into: A + dark:B (splitting B into words)
    const regex = /\$\{theme === ['"]light['"] \? (['"`])([^'"`]+)\1 : (['"`])([^'"`]+)\3\}/g;
    
    content = content.replace(regex, (match, q1, lightClasses, q2, darkClasses) => {
        const dClasses = darkClasses.split(/\s+/).map(c => {
            if (!c) return '';
            // If the dark class doesn't already have dark:, add it
            if (c.startsWith('dark:')) return c;
            // Also, some classes might be identical, we could filter them but tailwind doesn't care
            return `dark:${c}`;
        }).join(' ');
        
        return `${lightClasses} ${dClasses}`;
    });
    
    // Pattern 2: ${view === "categories" ? (theme === 'light' ? 'A' : 'B') : 'C'}
    // This is trickier. We can just replace the inner theme check
    const innerRegex = /\(theme === ['"]light['"] \? (['"`])([^'"`]+)\1 : (['"`])([^'"`]+)\3\)/g;
    content = content.replace(innerRegex, (match, q1, lightClasses, q2, darkClasses) => {
        const dClasses = darkClasses.split(/\s+/).map(c => {
            if (!c) return '';
            if (c.startsWith('dark:')) return c;
            return `dark:${c}`;
        }).join(' ');
        
        return `'${lightClasses} ${dClasses}'`;
    });

    // Replace instances where we might have leftover `bg-surface` with `bg-[#12141C]` since it's hardcoded to white
    content = content.replace(/dark:bg-surface/g, 'dark:bg-[#12141C]');
    
    if (content !== originalContent) {
        fs.writeFileSync(file, content, 'utf8');
        console.log(`Updated ${file}`);
        count++;
    }
}

console.log(`Finished updating ${count} files.`);
